addon/components/sl-menu.js

```diff
@@ -232,415 +94,555 @@ export default Ember.Component.extend({
-        if ( this.get( 'keyboardInUse' ) ) {
-            const currentIndex = this.get( 'currentRootNodeIndex' );
-
-            if ( currentIndex ) {
-                const query = 'a:contains("' +
-                    this.get( 'rootNode.menu.pages' )[ currentIndex ].label +
-                    '")';
-
-                this.$( query ).parent().removeClass( 'active' );
-            }
-
-            this.set( 'keyboardInUse', false );
-        }
-
-        this.$().addClass( 'active' );
-    },
-
-    /**
+     * mouseLeave event handler
+     *
      * @function
      * @returns {undefined}
      */
     mouseLeave() {
-        if ( this.get( 'isRoot' ) ) {
-            this.send( 'closeAll' );
-        } else if ( !this.get( 'rootNode' ).$().hasClass( 'showall' ) ) {
-            this.$().removeClass( 'active' );
-        }
+        this.send( 'hideAll' );
     },
 
     // -------------------------------------------------------------------------
     // Properties
 
     /**
-     * Currently active child
+     * Whether to show a menu item to display all sub-menus
      *
-     * @type {?Number}
-     */
-    activeChild: null,
-
-    /**
-     * Embedded Ember View representing the "Show All"
-     *
-     * @type {ember/View}
+     * @type {Boolean}
      */
-    AllView: SlMenuShowAllView,
+    allowShowAll: false,
 
     /**
-     * Collection of children items
+     * The array of menu items
      *
      * @type {?Object[]}
      */
-    children: null,
+    items: null,
 
     /**
-     * Array of classes to be added to the element's class attribute
+     * An of objects containing data about the selected states
      *
-     * @type {String[]}
+     * @private
+     * @type {?ember/Array}
      */
-    extraClassNames: [],
+    selections: null,
 
     /**
+     * Whether to show all the menu's sub-items
+     *
+     * @private
      * @type {Boolean}
      */
-    isRoot: true,
-
-    /**
-     * @type {?Object[]}
-     */
-    keyEvents: null,
+    showingAll: false,
 
-    /**
-     * @type {Boolean}
-     */
-    keyHandler: false,
+    // -------------------------------------------------------------------------
+    // Observers
 
     /**
-     * Is the menu being interacted with via the keyboard?
+     * Initialize any computed properties that need setup
      *
-     * @type {Boolean}
+     * @function
+     * @returns {undefined}
      */
-    keyboardInUse: false,
+    initialize: Ember.on(
+        'init',
+        function() {
+            this.set( 'selections', new Ember.A() );
+        }
+    ),
 
     /**
-     * Whether "Show All" icon and functionality is enabled
-     *
-     * This is a string representaton of a boolean state.
+     * Setup the stream actions bindings
      *
-     * @type {String}
+     * @function
+     * @returns {undefined}
      */
-    showAll: 'false',
+    setupStreamActions: Ember.on(
+        'init',
+        function() {
+            const stream = this.get( 'stream' );
```

Comment:
No, that's fine, as long as it's not something passed in via `{{sl-menu stream="myStream"}}`